controllers/componentversion_controller.go

```diff
@@ -92,18 +93,77 @@ func (r *ComponentVersionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		}, fmt.Errorf("attempted to verify component, but the digest didn't match")
 	}
 
-	return r.reconcile(ctx, component)
+	// reconcile the version before calling reconcile func
+	update, version, err := r.checkVersion(ctx, component)
+	if err != nil {
+		return ctrl.Result{
+			RequeueAfter: component.GetRequeueAfter(),
+		}, fmt.Errorf("failed to check version: %w", err)
+	}
+
+	if !update {
+		return ctrl.Result{}, nil
+	}
+
+	return r.reconcile(ctx, component, version)
+}
+
+func (r *ComponentVersionReconciler) checkVersion(ctx context.Context, obj *v1alpha1.ComponentVersion) (bool, string, error) {
+	log := log.FromContext(ctx).WithName("ocm-component-version-reconcile")
+
+	// get given semver constraint
+	constraint, err := semver.NewConstraint(obj.Spec.Version.Semver)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to parse given semver constraint: %w", err)
+	}
+
+	// get current reconciled version
+	reconciledVersion := "0.0.0"
+	if obj.Status.ReconciledVersion != "" {
+		reconciledVersion = obj.Status.ReconciledVersion
+	}
+	current, err := semver.NewVersion(reconciledVersion)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to parse reconciled version: %w", err)
+	}
+	log.V(4).Info("current reconciled version is", "reconciled", current.String())
+
+	// compare given version and current reconciled version
+	if constraint.Check(current) {
```

Comment:
Hmm, I'm not sure about this one. For example, if my constraint is `>=v1-0`, this will satisfy a lot of versions. So the latest will be okay as well and might be different from the current one, and we wouldn't get it because the current version also satisfies this constraint.